src/Symfony/Component/Form/Tests/SimpleFormTest.php

```diff
@@ -1036,15 +1037,17 @@ public function testPostSubmitDataIsNullIfInheritData()
 
     public function testSubmitIsNeverFiredIfInheritData()
     {
-        $test = $this;
+        $called = 0;
         $form = $this->getBuilder()
-            ->addEventListener(FormEvents::SUBMIT, function (FormEvent $event) use ($test) {
-                $test->fail('The SUBMIT event should not be fired');
+            ->addEventListener(FormEvents::SUBMIT, function (FormEvent $event) use (&$called) {
+                ++$called;
             })
             ->setInheritData(true)
             ->getForm();
 
         $form->submit('foo');
+
+        $this->assertSame(0, $called);
```

Comment:
I would add a message here (``'The SUBMIT event should not be fired.'``) to make the failure easier to understand